packages/qdrant-loader/src/qdrant_loader/core/state/state_change_detector.py

```
@@ -113,6 +113,77 @@ async def detect_changes(
+                        uri=uri,
+                        content_hash=record.content_hash,  # type: ignore
+                        updated_at=record.updated_at,  # type: ignore
+                    )
+            except Exception as e:
+                self.logger.error(
+                    "Error loading existing document state for batch classification",
+                    source_type=source_type,
+                    source=source,
+                    error=str(e),
+                    error_type=type(e).__name__,
+                )
```

Ruff (0.15.14)

[warning] Use `logging.exception` instead of `logging.error`

Replace with `exception`

(TRY400)